.pre-commit-config.yaml

```diff
@@ -10,3 +10,10 @@ repos:
     hooks:
     - id: black
       language_version: python3
+# Mypy
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: 'v1.11.2'  
+    hooks:
+    - id: mypy 
+      name: mypy 
+      additional_dependencies: [types-PyYAML==6.0.12.20240808]
```

Comment:
What is the effect of this?

Reply:
Adds a mypy check to pre-commit - where type annotations are added, this will reject changes if they're incorrect.

Reply:
N.B if there aren't any type annotations added, this won't affect anything.